mesh/pmesh.cpp

```diff
@@ -1739,111 +1742,76 @@ void ParMesh::GetSharedTriCommunicator(int ordering,
 
 void ParMesh::MarkTetMeshForRefinement(const DSTable &v_to_v)
 {
-   Array<int> order;
-   GetEdgeOrdering(v_to_v, order); // local edge ordering
+   Array<double> lengths;
+   GetEdgeLengths(v_to_v, lengths);
 
-   // create a GroupCommunicator on the shared edges
+   // create a GroupCommunicator over shared edges
    GroupCommunicator sedge_comm(gtopo);
    GetSharedEdgeCommunicator(0, sedge_comm);
 
-   Array<int> sedge_ord(shared_edges.Size());
-   Array<Pair<int,int> > sedge_ord_map(shared_edges.Size());
-   for (int k = 0; k < shared_edges.Size(); k++)
+   // communicate the local index of each shared edge from the group master to
+   // other ranks in the group
+   Array<int> sedge_master_rank(shared_edges.Size());
+   Array<int> sedge_master_index(shared_edges.Size());
+   for (int i = 0; i < group_sedge.Size(); i++)
+   {
+      int rank = gtopo.GetGroupMasterRank(i+1);
```

Comment:
`const int rank`

Reply:
Resolved in [e14459b](https://github.com/mfem/mfem/pull/3847/commits/e14459bfd057b1be1ba2ffa5ee19b97ebb085990)